packages/babel-parser/src/tokenizer/index.js

```diff
@@ -971,14 +971,23 @@ export default class Tokenizer extends LocationParser {
 
   readNumber(startsWithDot: boolean): void {
     const start = this.state.pos;
-    let octal = this.input.charCodeAt(start) === charCodes.digit0;
     let isFloat = false;
     let isBigInt = false;
 
     if (!startsWithDot && this.readInt(10) === null) {
       this.raise(start, "Invalid number");
     }
-    if (octal && this.state.pos == start + 1) octal = false; // number === 0
+    let octal =
+      this.state.pos - start >= 2 &&
+      this.input.charCodeAt(start) === charCodes.digit0;
+    if (octal) {
+      if (this.state.strict) {
+        this.raise(start, "Invalid number");
```

Comment:
`Legacy octal literals are not supported in strict mode`?

Reply:
Yeah, this is better.

Is "Legacy" needed?

> Octal literals are not allowed in strict mode.

Reply:
Makes sense. Changed it.